packages/visual-settings/src/helpers.ts

```diff
@@ -288,36 +296,42 @@ function buildEnumerationObject(
 	instances: powerbi.VisualObjectInstance[]
 ) {
 	'use strict'
-	const adapted = convertValueToPBI(
-		settingsObj,
-		setting,
-		dataView,
-		includeHidden
-	)
-	if (adapted) {
-		const { objName, propName } = getPBIObjectNameAndPropertyName(setting)
-		let value = adapted.adaptedValue
-		value = value && value.forEach ? value : [value]
-		value.forEach((n: any) => {
-			const isVisualInstance = !!(n && n.properties)
-			let instance = n as powerbi.VisualObjectInstance
-			if (isVisualInstance) {
-				instance = merge(instance, {
-					objectName: objName
-				})
-				if (
-					typeof instance.displayName === 'undefined' ||
-					instance.displayName === null
-				) {
-					// tslint:disable-line
-					instance.displayName = '(Blank)'
+	const { readOnly, enumerable } = setting.descriptor
+	if (readOnly !== true && enumerable !== false) {
+		const adapted = convertValueToPBI(
+			settingsObj,
+			setting,
+			dataView,
+			includeHidden
+		)
+		// There is no reason to run the setting conversion if it shouldn't even be enumerated
+		if (adapted) {
+			const { objName, propName } = getPBIObjectNameAndPropertyName(
+				setting
+			)
+			let value = adapted.adaptedValue
+			value = value && value.forEach ? value : [value]
+			value.forEach((n: any) => {
+				const isVisualInstance = !!(n && n.properties)
+				let instance = n as powerbi.VisualObjectInstance
+				if (isVisualInstance) {
+					instance = merge(instance, {
```

Comment:
Can you extract a new helper function here? The result should look something like 
```
  instances.push(convertInstance(instance))
```